src/aks-preview/azext_aks_preview/_roleassignments.py

```diff
@@ -0,0 +1,136 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import time
+import uuid
+from azure.graphrbac.models import GetObjectsParameters
+from knack.log import get_logger
+from knack.util import CLIError
+from msrestazure.azure_exceptions import CloudError
+from ._client_factory import get_auth_management_client, get_graph_rbac_management_client
+
+logger = get_logger(__name__)
+
+
+def _get_object_stubs(graph_client, assignees):
+    params = GetObjectsParameters(include_directory_object_references=True,
+                                  object_ids=assignees)
+    return list(graph_client.objects.get_objects_by_object_ids(params))
+
+
+def resolve_object_id(cli_ctx, assignee):
+    client = get_graph_rbac_management_client(cli_ctx)
+    result = None
+    if assignee.find('@') >= 0:  # looks like a user principal name
+        result = list(client.users.list(
+            filter="userPrincipalName eq '{}'".format(assignee)))
+    if not result:
+        result = list(client.service_principals.list(
+            filter="servicePrincipalNames/any(c:c eq '{}')".format(assignee)))
+    if not result:  # assume an object id, let us verify it

```

Comment:
result is None

Reply:
Hi Andy, a lot of this code is not mine. It was simply moved around for organization purposes and to avoid circular dependencies. I ensured everything still works, but did not change any code functionality except for functions that were directly related to addons (and there were only 2 of them). I'm not sure it would be safe for me to change the functionality of these other functions without knowing there purpose.

This one seems easy enough to add in, however, if you want me to, though it might break things if result could simply be a False value for whatever reason.